src/deposit/views.py

```
@@ -59,12 +122,15 @@ def deposit(request):
 
     # Verify the optional request arguments.
     verify_optional_args = _verify_optional_args(request)
-    if verify_optional_args is not None:
+    if verify_optional_args:
         return verify_optional_args
 
     # TODO: Check if the provided Stellar account exists, and if not, create it.
 
+    # Construct interactive deposit pop-up URL.
+    transaction_id = _create_transaction(asset_code, stellar_account)
```

We should probably validate if a transaction with the given id already exist. If it does, we should throw an error, instead of overwriting the existing one.

---

Good point. Previously, I was calling the `uuid.uuid4()` function separately, and then passing that as the `id` when creating a `Transaction` object. However, setting `uuid.uuid4` as a default function on the `id` field of `Transaction` means Django will use that function to a `Transaction` object ID. So, I am no longer externally calling that function, and just letting Django do the work. This should solve this potential issue.

---

Ah, never mind, I don't think the above approach quite makes sense in the context of [this comment](https://github.com/stellar/stellar-anchor-server/pull/45#discussion_r313440593): it would result in the premature transaction creation mentioned here.